sources/Builders/FromArray.php

```diff
@@ -0,0 +1,64 @@
+<?php
+declare(strict_types=1);
+
+namespace Ciebit\News\Builders;
+
+use Ciebit\News\News;
+use Ciebit\News\Status;
+use Ciebit\Stories\Builders\FromArray as StoryBuilder;
+use Ciebit\Files\Images\Builders\FromArray as ImageBuilder;
+use DateTime;
+
+class FromArray implements Builder
+{
+    private $data; #:array
+
+    public function setData(array $data): self
+    {
+        $this->data = $data;
+        return $this;
+    }
+
+    public function build(): News
+    {
+        if (
+            ! is_array($this->data) OR
+            ! isset($this->data['story_id'])
+        ) {
+            throw new Exception('ciebit.news.builders.invalid', 3);
+        }
+        $story = (new StoryBuilder)->setData([
+            'title' => $this->data['story_title'],
```

Comment:
Aqui você não fez a checagem dessa chave.

Vendo agora, me limitei a facilitar o trabalho pensando apenas nos dados vindos via banco de dados, o correto mesmo é que os dados venham com a seguinte estrutura:

```php
$news = [
    'id' => 1
    'story' => []
    'imagem' => [],
    'status' => 2
];
```
Assim a checagem se resume a verificação de um array e a passagem para seus respectivos *builders*

Reply:
@johnnysaymon , creio que não entendi a mensagem que você quis passar

Reply:
@maikenegreiros  O builder está esperando um array com a seguinte estrutura:

```
$news = [
    'id' => 1,
    'story_id' => 2,
    'story_title' => 'Title New',
    // ...
    'image_id' => 5,
    'image_name' => 'Name of Image',
    // ...
    'status' => 2
]
```
Estou sugerindo que ele passe a esperar a estrutura que mencionei no comentário https://github.com/ciebit/news/pull/1#discussion_r208377418

Reply:
Isso quer dizer que da linha 30 a 39 seria substituído apenas por $story = (new StoryBuilder)->setData($this->data['story']);?